### ISSUE TITLE
networkmanager shows virtual interfaces

### ISSUE THREAD

**Issue by DamienCassou** [NixOS/nixpkgs#7912]
Starting from Gnome 3.16, I see 2 ethernet devices in Gnome:

![nixos-ethernet](https://cloud.githubusercontent.com/assets/217543/7725532/df57769e-fefb-11e4-8b6e-188e5b399bc3.png)

This also affects the network settings panel.

**Comment by lethalman** [comment]
Can you paste your `ifconfig` ?

**Comment by vcunat** [comment]
Could it be some virtual interface? (discussed around https://github.com/NixOS/nixpkgs/pull/7357#issuecomment-92288377)

**Comment by DamienCassou** [comment]
```
enp1s0f0  Link encap:Ethernet  HWaddr XXX
          inet addr:XXX  Bcast:XXX  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:249407 errors:0 dropped:5 overruns:0 frame:0
          TX packets:85615 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:319675454  TX bytes:7707591
          Interrupt:16

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:20166 errors:0 dropped:0 overruns:0 frame:0
          TX packets:20166 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:13185584  TX bytes:13185584

vboxnet0  Link encap:Ethernet  HWaddr 0A:00:27:00:00:00
          inet addr:192.168.XXX  Bcast:0.0.0.0  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0  TX bytes:648
```

**Comment by lethalman** [comment]
I see, so it's probably that vboxnet0. You're lucky it's only one at least :P This is something upstream has to solve.

**Comment by vcunat** [comment]
It's solved on NM master, and they promised to backport to 1.0 after 1.0.2. However, 1.0.2 was released over two weeks ago and I don't see it done yet. https://bugzilla.gnome.org/show_bug.cgi?id=731014#c72

**Comment by DamienCassou** [comment]
I've asked them again to do the backport. Can't we use 1.0.2?

**Comment by lethalman** [comment]
@DamienCassou sure feel free to update and test it :) that's a lot of recompilation

**Comment by DamienCassou** [comment]
@lethalman see #7913

**Comment by DamienCassou** [comment]
I've worked on updating nixpkgs to network-manager 1.0.2 in a dedicated branch https://github.com/DamienCassou/nixpkgs/tree/update-networkmanager-1.0.2. Everything compiles and runs fine. But the bug doesn't seem solved as I still see 2 network interfaces.

**Comment by lethalman** [comment]
@DamienCassou thanks for the effort. I think we can merge the NM update anyway. @vcunat should this go in staging? Will rebuild lot of gnome plus webkit, chromium ecc...

**Comment by vcunat** [comment]
It's probably better, although I have no strong opinion on this case.

**Comment by lethalman** [comment]
@DamienCassou ok I think you can go ahead and push to master :) Thanks.

**Comment by DamienCassou** [comment]
ok, I pushed to master. The issue discussed here still exists though.

**Comment by vcunat** [comment]
NM users ping: 1.0.4 was released and seems to contain the fix for this, would you test it?

**Comment by lethalman** [comment]
I cannot reproduce this problem even with current NM, so perhaps @DamienCassou ?

**Comment by DamienCassou** [comment]
I removed virtualbox as I don't really use it anymore.

**Comment by fazo96** [comment]
I have the same exact issue on unstable right now (NetworkManager still at 1.0.2). `vboxnet0` is shown as "Ethernet". I'll see if I have the time to update the NetworkManager expression to the latest version.

**Comment by lethalman** [comment]
Confirming, but at least only few virtual interfaces are shown and not thousands like before. This is still an upstream issue I believe.

Anybody can confirm this also happens with other distros?

**Comment by vcunat** [comment]
You should try with 1.0.4 as mentioned above &ndash; upstream claims it contains backport of hiding (some of) virtual interfaces (I don't remember much anymore).

**Comment by ktosiek** [comment]
I can still see vboxnet0 in 1.0.6 (and I'm not even sure it's a bug with a sensible fix - I think we should tell NM to hide the interfaces we create ;-)). Looking at https://bugzilla.gnome.org/show_bug.cgi?id=731014 the fix is targeted for NM 1.2

**Comment by vcunat** [comment]
The fix was backported upstream to the 1.0 branch, right after 1.0.2 release IIRC, so it should be contained already.

**Comment by garbas** [comment]
#10862 solves this and i just merged it. please reopen if #10862 does not fixes it
